src/main/java/mate/jdbc/dao/impl/DriverDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createDriverStatement =
+                        connection.prepareStatement(INSERT_DRIVER_REQUEST,
+                             Statement.RETURN_GENERATED_KEYS);) {
+            createDriverStatement.setString(1, driver.getName());
+            createDriverStatement.setString(2, driver.getLicenseNumber());
+            createDriverStatement.executeUpdate();
+            ResultSet generatedKeys = createDriverStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                driver.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(CANT_CREATE_MESSAGE, e);
+        }
+        return driver;
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateDriverStatement =
+                        connection.prepareStatement(UPDATE_DRIVER_REQUEST,
+                             Statement.RETURN_GENERATED_KEYS);) {
+            updateDriverStatement.setString(1, driver.getName());
+            updateDriverStatement.setString(2, driver.getLicenseNumber());
+            updateDriverStatement.setObject(3, driver.getId());
+            updateDriverStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException(CANT_UPDATE_MESSAGE, e);
+        }
+        return driver;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteDriverStatement =
+                         connection.prepareStatement(DELETE_DRIVER_REQUEST,
+                             Statement.RETURN_GENERATED_KEYS);) {
+            deleteDriverStatement.setLong(1, id);
+            return deleteDriverStatement.executeUpdate() >= 1;
+        } catch (SQLException e) {
+            throw new RuntimeException(CANT_DELETE_MESSAGE, e);
+        }
+    }
+
+    @Override
+    public Optional<Driver> get(Long id) {
+        Driver driver = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getDriverStatement =
+                        connection.prepareStatement(GET_DRIVER_REQUEST,
+                             Statement.RETURN_GENERATED_KEYS);) {
+            getDriverStatement.setLong(1, id);
+            ResultSet resultSet = getDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                driver = new Driver();
+                String name = resultSet.getString(NAME);
+                String licenceNumber = resultSet.getString(LICENCE_NUMBER);
+                driver.setId(id);
+                driver.setName(name);
+                driver.setLicenseNumber(licenceNumber);
+            }
```

Comment:
better to use if instead of while, because you must find only one driver. and use a separate method as described above

Reply:
fixed